lntest/itest/lnd_signer_test.go

```diff
@@ -190,6 +196,195 @@ func runDeriveSharedKey(t *harnessTest, alice *lntest.HarnessNode) {
 	assertErrorMatch("use either raw_key_bytes or key_index", req)
 }
 
+// testSignOutputRaw makes sure that the SignOutputRaw RPC can be used with all
+// custom ways of specifying the signing key in the key descriptor/locator.
+func testSignOutputRaw(net *lntest.NetworkHarness, t *harnessTest) {
+	runSignOutputRaw(t, net, net.Alice)
+}
+
+// runSignOutputRaw makes sure that the SignOutputRaw RPC can be used with all
+// custom ways of specifying the signing key in the key descriptor/locator.
+func runSignOutputRaw(t *harnessTest, net *lntest.NetworkHarness,
+	alice *lntest.HarnessNode) {
+
+	ctxb := context.Background()
+	ctxt, cancel := context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+
+	// For the next step, we need a public key. Let's use a special family
+	// for this. We want this to be an index of zero.
+	const testCustomKeyFamily = 44
+	keyDesc, err := alice.WalletKitClient.DeriveNextKey(
+		ctxt, &walletrpc.KeyReq{
+			KeyFamily: testCustomKeyFamily,
+		},
+	)
+	require.NoError(t.t, err)
+	require.Equal(t.t, int32(0), keyDesc.KeyLoc.KeyIndex)
+
+	targetPubKey, err := btcec.ParsePubKey(keyDesc.RawKeyBytes)
+	require.NoError(t.t, err)
+
+	// First, try with a key descriptor that only sets the public key.
+	assertSignOutputRaw(
+		ctxt, t, net, alice, targetPubKey, &signrpc.KeyDescriptor{
```

Comment:
nit: not that ideally we'd need separate timeout contexts for all these calls as ctxt has an absolute timeout.

Reply:
Interestingly the failure in the `etcd` itest is exactly due to this timeout context.

Reply:
Ah, good catch! I assumed this would be done in a few seconds at most. But you're right, the remote signer case with `etcd` does take a bit longer. I added multiple timeout contexts.